src/kOS/Module/kOSProcessor.cs

```diff
@@ -818,19 +871,26 @@ private void ProcessorModeChanged()
         {
             switch (ProcessorMode)
             {
-            case ProcessorModes.READY:
-                shared.VolumeMgr.SwitchTo(SafeHouse.Config.StartOnArchive ? (Volume)Archive : HardDisk);
-                if (shared.Cpu != null) shared.Cpu.Boot();
-                if (shared.Interpreter != null) shared.Interpreter.SetInputLock(false);
-                if (shared.Window != null) shared.Window.IsPowered = true;
-                break;
-
-            case ProcessorModes.OFF:
-            case ProcessorModes.STARVED:
-                if (shared.Interpreter != null) shared.Interpreter.SetInputLock(true);
-                if (shared.Window != null) shared.Window.IsPowered = false;
-                if (shared.BindingMgr != null) shared.BindingMgr.UnBindAll();
-                break;
+                case ProcessorModes.READY:
+                    if (SafeHouse.Config.StartOnArchive)
+                    {
+                        shared.VolumeMgr.SwitchTo(Archive);
+                    }
+                    else
+                    {
+                        shared.VolumeMgr.SwitchTo(HardDisk);
+                    }
```

Comment:
I changed the logic in the first commit, when I had to [here](https://github.com/KSP-KOS/KOS/pull/1686/commits/517510c62b5bc6433ed4ee4a3a4785c9cd4aaafd#diff-7be643b4883b5a4b8146fc3687dccf1cR823).  When I changed the higher level logic to use a `VolumePath` instead of `GlobalPath` I no longer needed to set the volume of the `GlobalPath` in this method, and I simply forgot to simplify it back to a one line expression.  Feel free to revert it.